### PR TITLE
Add support for setup/teardown commands

### LINKED ISSUE #25: Feature Request - cleanup between runs
I'd like a flag to `bench` to specify a command to run between invocations of the command under test, without including it in the timing.  Something like `bench --setup 'make clean' 'make'`.

Right now, I can either `make clean && time make` and compile stats of multiple runs myself, or `bench 'make clean && make'` which adds any cleanup time to the measured runtime.

### PR DESCRIPTION
**Pull request by Gabriel439**
Fixes #25

### CODE REVIEW

File: src/Main.hs [hunk 1]
```diff
@@ -24,7 +24,25 @@ import qualified Turtle
...
+        (   Options.Applicative.long "before"
+        <>  Options.Applicative.help "Specify a command to run before each run of the benchmark"
+        <>  Options.Applicative.metavar "command"
+        )
+
+after :: Parser Text
+after =
+    Options.Applicative.strOption
+        (   Options.Applicative.long "after"
+        <>  Options.Applicative.help "Specify a command to run before after run of the benchmark"
```
Comment (bergey) [inline src/Main.hs]: typo, should say "after each run"

**Comment by bergey** [comment]
Thanks for adding this feature!  Unfortunately, it's not yet doing what I hoped for.

`bench 'stack clean; stack build'` gives `time                 6.467 s`
`bench --before 'stack clean' 'stack build'` gives `time                 381.9 ms`

I would expect the latter to be about the same as the former, maybe ~0.3 s less because it doesn't include the cleanup time.  I think < 400 ms means that most runs of `stack build` find that nothing needs to be rebuilt.

Looking at the Criterion docs, maybe `perRunEnvWithCleanup` would do what I want?  I can test that later this week.  I'm not sure how likely it is that other users of `bench` would want the behavior in this PR, or other variations.

**Comment by Gabriel439** [comment]
@bergey: I will use `perRunEnvWithCleanup`.  I just misunderstood what `envWithCleanup` did and I think most users will want the clean up between each run

**Comment by bergey** [comment]
I had to make [a few more changes](https://github.com/bergey/bench/commit/ce15d5caf105390a4610c9472c84ae666346910d) to get this to compile.  With those changes, it does what I hoped.

**Comment by Gabriel439** [comment]
Whoops!  I thought I had built the branch before pushing.  Thanks for the fix, which I've included now

**Pull request merged**